sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1288,3 +1288,66 @@ def analyzer(doc):
 
     with pytest.raises(Exception, match="testing"):
         Estimator(analyzer=analyzer, input='file').fit_transform([f])
+
+
+@pytest.mark.parametrize(
+    'Vectorizer',
+    [CountVectorizer, HashingVectorizer, TfidfVectorizer]
+)
+def test_unused_parameters_warn(Vectorizer):
```

Comment:
It might also be neat if we parameterized the test with parameters like (unused_name, unused_value, overriding_name, overriding_value) and just match the part of the message that says "{unused_name}' will not be used.

Reply:
Done . Had to add a condition to check for ngram_range != (1, 1)  in the parameters and the warning checks . 
`        if (self.ngram_range != (1, 1) and self.ngram_range is not None
                and callable(self.analyzer)):
            warnings.warn("The parameter 'ngram_range' will not be used"
                          " since 'analyzer' is callable'")`